pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader_test.go

```diff
@@ -347,6 +332,52 @@ type fakeChainReader struct {
 	lock        sync.Mutex
 }
 
+type fakeChainWriter struct {
+	types.ChainWriter
+	cr *fakeChainReader
+}
+
+func (f *fakeChainWriter) SubmitTransaction(ctx context.Context, contractName, method string, args any, transactionID string, toAddress string, meta *types.TxMeta, value *big.Int) error {
+	switch method {
+	case "addTestStruct":
```

Comment:
Can we move these into constants so that they can be reused in chain reader configs for the tests in the relayers. If you think its out of scope then just a TODO with a ticket number